### DISCUSSION TITLE
Seed for RandomLinkSplit in torch_geometric.transforms

### DISCUSSION THREAD

**Discussion by songsong0425** [pyg-team/pytorch_geometric#7373] [Ideas]
Before starting, thank you for your effort on the graph-based deep learning model.
While running the PyG code, I needed to set seed to maintain the consistency of training, validation, and test set for a fair comparison.
However, I couldn't find the seed-related option in `RandomLinkSplit`.
For cases similar to this, how about adding an option such as `seed=1234` like `random.seed()`?
Or if there was any related option in `RandomLinkSplit`, please notify me.
Thank you for your reading!

**Comment by bartvdbraak** [comment]
Hi @songsong0425,

Regarding your suggestion to add a seed-related option to RandomLinkSplit, I think it's a great idea. In fact, I tried adding this option to the codebase. You can find my commits related to this change in the following branch on the GitHub repository:

https://github.com/bartvdbraak/pytorch_geometric/commit/dc235f0a01ea29bd1ff61ac8d753aec57b7f5753

In this branch, you'll see that I added a `seed` argument to the constructor of the RandomLinkSplit class, which allows you to specify a seed value for the random split. You can use this option to ensure that the split is consistent across runs.

**Reply by songsong0425** [reply]
Thank you for your reply and modification. I confirmed the change and it looks great. Then, can you update the PyG docs(https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.transforms.RandomLinkSplit.html#torch_geometric.transforms.RandomLinkSplit) adding this option?
Sorry to bother you, but it will be helpful who use PyG.
